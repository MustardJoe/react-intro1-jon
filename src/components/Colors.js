import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, hex }) {
  const color = {
    name,
    hex
  };
  const fields = Object.keys(color)
    .map(key => {
      return (
        <>
          <dt>{key}</dt>
          <dd style={{ color: hex }}></dd>
        </>
      );
    });


  return (
    <dl>
      {fields}
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired,
};

export default Color;
