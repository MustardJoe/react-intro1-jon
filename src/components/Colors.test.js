import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';
import { exportAllDeclaration } from '@babel/types';

describe('It is the test for colors component', () => {
  it('creates a list of I think 5 maybe 4 4 or 5 colors', () => {
    const colors = [
      { name: 'red', rgb: { red: 255, green: 0, blue: 0 } },
      { name: 'green', rgb: { red: 0, green: 255, blue: 0 } },
      { name: 'orange', rgb: { red: 255, green: 165, blue: 0 } },
      { name: 'hotpink', rgb: { red: 255, green: 105, blue: 180 } }
    ];

    const wrapper = shallow(<Colors colors={colors} />);
    expect(wrapper).toMatchSnapshot();
  });
});
