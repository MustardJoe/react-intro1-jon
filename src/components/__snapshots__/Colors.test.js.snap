// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It is the test for colors component creates a list of I think 5 maybe 4 4 or 5 colors 1`] = `
<ul>
  <li
    key="red"
  >
    <Color
      name="red"
      rgb={
        Object {
          "blue": 0,
          "green": 0,
          "red": 255,
        }
      }
    />
  </li>
  <li
    key="green"
  >
    <Color
      name="green"
      rgb={
        Object {
          "blue": 0,
          "green": 255,
          "red": 0,
        }
      }
    />
  </li>
  <li
    key="orange"
  >
    <Color
      name="orange"
      rgb={
        Object {
          "blue": 0,
          "green": 165,
          "red": 255,
        }
      }
    />
  </li>
  <li
    key="hotpink"
  >
    <Color
      name="hotpink"
      rgb={
        Object {
          "blue": 180,
          "green": 105,
          "red": 255,
        }
      }
    />
  </li>
</ul>
`;
